src/selectors/createSelectors.ts

```diff
@@ -0,0 +1,44 @@
+import {createSelectorCreator} from './createSelector';
+import {createSelectorByCssCreator} from './createSelectorByCss';
+import {createDefaultCustomMethods} from './defaultCustomMethods';
+import {htmlElementSelectorCreator} from './htmlElementSelector';
+import {locatorIdSelectorCreator} from './locatorIdSelector';
+
+import type {
+  CreateSelectorsOptions,
+  GetTestAttrNameFn,
+  SelectorCustomMethods,
+} from '../types/internal';
+
+const createSelectorsWithCustomMethods = <CustomMethods extends SelectorCustomMethods = {}>(
+  getTestAttrName: GetTestAttrNameFn,
+  // force `this` to be Selector
+  customMethods?: CustomMethods,
```

Comment:
`nitpick`: I understand that the `customMethods` argument (and the `CustomMethods` type parameter) are left over from the version in which we intended to allow the `Selector` extension with custom methods.
But now no custom methods and no generics are expected; we know the type of `createSelectors` in advance.
However, the `CustomMethods` type parameter is still passed in dozens of places in the code.

Let's remove the `customMethods` argument and (especially) the `CustomMethods` type parameter from everywhere, this will greatly simplify the implementation code.
```suggestion
```

Reply:
if you 100% sure that it will not be needed in the future, i can remove it

Reply:
Yes, since we decided not to make the `Selector` type changeable within projects (so as not to complicate the understanding of the code), the `CustomMethods` type parameter can be deleted.

In the projects, developers will have one explicit layer left to customize the behavior of elements — `pageObject`-s.

And inside `e2ed`, even without `CustomMethods`, we will still be able to further expand the `Selector`.